homework/index.js

```diff
@@ -30,15 +32,95 @@
     return elem;
   }
 
+  function clearContainer(container) {
+    while (container.firstChild) {
+      container.removeChild(container.firstChild);
+    }
+  }
+
+  function renderError(err) {
+    const root = document.getElementById('root');
+    createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+  }
+
+  function addRow(tbody, label, value = '') {
+    const row = createAndAppend('tr', tbody);
+    createAndAppend('td', row, { text: `${label} :`, class: 'label' });
+    createAndAppend('td', row, { text: value });
+    return row;
+  }
+
+  function createInfoList(repository, container) {
+    clearContainer(container);
+    const table = createAndAppend('table', container);
+    const tbody = createAndAppend('tbody', table);
+    const firstRow = addRow(tbody, 'Repository');
+    createAndAppend('a', firstRow.lastChild, {
+      href: repository.html_url,
+      target: '_blank',
+      text: repository.name,
+    });
+    addRow(tbody, 'Description:', repository.description);
+    addRow(tbody, 'Forks:', repository.forks);
+    addRow(tbody, 'Updated:', new Date(repository.updated_at).toLocaleString());
+  }
+
+  function createContributorsList(repository, ul) {
+    clearContainer(ul);
+    fetchJSON(repository.contributors_url)
+      .then(contributors =>
+        contributors.forEach(contributor => {
+          const li = createAndAppend('li', ul);
+          const a = createAndAppend('a', li, {
+            href: repository.html_url,
+            target: '_blank',
+            id: 'link_contributions',
+          });
+          const div = createAndAppend('div', a);
+          createAndAppend('img', div, {
+            class: 'contributor_avatar',
+            src: contributor.avatar_url,
+            alt: contributor.login,
+          });
+          createAndAppend('p', div, { id: 'p1', text: contributor.login });
+          createAndAppend('p', div, { id: 'p2', text: contributor.contributions });
+        }),
+      )
+      .catch(err => renderError(err));
+  }
+
+  function startPage(repository, listContainer, ul) {
+    createContributorsList(repository, listContainer);
+    createInfoList(repository, ul);
+  }
+
   function main(url) {
-    fetchJSON(url, (err, repositories) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
-        return;
-      }
-      createAndAppend('pre', root, { text: JSON.stringify(repositories, null, 2) });
+    const root = document.getElementById('root');
+    const header = createAndAppend('header', root);
+    const section = createAndAppend('section', root);
+    createAndAppend('h1', header, { text: 'HYF Repositories' });
+    const table = createAndAppend('div', section, { id: 'repo_information_table_container' });
```

Comment:
The name you chose for this variable suggests that you are creating a `table` element. But you are creating a `div`. Therefore this name is a bit confusing.